### ISSUE TITLE
PkgDev.register failed with UndefVarError

### ISSUE THREAD

**Issue by yuyichao** [JuliaLang/PkgDev.jl#18]
Looks like it is refering to a function `normalize_url` that doesn't exist in `Base.LibGit2`.

```
julia> PkgDev.register("LibArchive")
ERROR: UndefVarError: normalize_url not defined
 in register at /usr/share/julia/site/v0.5/PkgDev/src/entry.jl:192
```

**Comment by tkelman** [comment]
Once we merge https://github.com/JuliaLang/julia/pull/11312, it should exist in `Pkg.Cache`? I'm surprised the tests didn't hit that.

**Comment by amitmurthy** [comment]
This issue remains even after https://github.com/JuliaLang/julia/pull/11312 has been merged.

**Comment by sfchen** [comment]
Also has this issue

**Comment by tkelman** [comment]
Try with `Pkg.checkout("PkgDev", "omm/ref")`, see https://github.com/JuliaLang/PkgDev.jl/pull/22 - just needs a test to be written then we can merge.

**Comment by samuel-massinon-invenia** [comment]
I also am having this issue. 
```
julia> PkgDev.register("VirtualArrays")
ERROR: UndefVarError: normalize_url not defined
 in register at /Users/samuelmassinon/.julia/v0.5/PkgDev/src/entry.jl:192
 in register at /Users/samuelmassinon/.julia/v0.5/PkgDev/src/PkgDev.jl:29
 in eval at ./boot.jl:265
```

**Comment by tkelman** [comment]
I merged #22, but we do still need to figure out how to test that on CI.
